FP_analysis_of_reviews_from_Amazon/test_common_functions.py

```diff
@@ -0,0 +1,84 @@
+from unittest import TestCase, main
+from common_functions import (
+    run_func_parallel,
+    get_apps_scores,
+    get_dict_of_apps_with_score,
+)
+
+class TestRunFuncParallel(TestCase):
+    def setUp(self):
+        self.data1 = ["a", "b", "c"]
+        self.data2 = [0]
+    def test_run_func_parallel(self):
+        self.assertEqual(run_func_parallel(len, [self.data1, self.data2]), [3, 1])
+
+class TestGetAppsScores(TestCase):
+    def setUp(self):
+        self.data = [
+            {"asin": "1", "overall": 1.0},
+            {"asin": "2", "overall": 5.0},
+            {"asin": "1", "overall": 5.0},
+            {"asin": "2", "overall": 4.0},
+            {"asin": "3", "overall": 4.0},
+        ]
+    def test_get_apps_scores(self):
+        self.assertEqual(
+            get_apps_scores(self.data), {"1": [1.0, 5.0], "2": [5.0, 4.0], "3": [4.0]}
+        )
+    def test_get_apps_scores_with_overall_exc(self):
+        self.data[0].pop("overall")
+        self.assertEqual(
+            get_apps_scores(self.data), {"1": [5.0], "2": [5.0, 4.0], "3": [4.0]}
+        )
+    def test_get_apps_scores_without_overall(self):
+        self.data[0].pop("overall")
```

Comment:
Mmm, can we just create a new dataset? Don't put extra logic into the tests if that's possible

Reply:
I thought that it is important to show differences with initial data to create new subtests. But of cause I fixed and created new dataset everywhere. Done.